torii/platform/vendor/xilinx.py

```diff
@@ -110,6 +110,29 @@ class XilinxPlatform(TemplatedPlatform):
 
 	Available overrides:
 		* ``add_constraints``: inserts commands in XDC file.
+
+	.. rubric:: yosys_nextpnr toolchain
+
+	Required tools:
+		* ``yosys``
+		* ``nextpnr-xilinx``
+		* ``fasm2frames``
+		* ``xc7frames2bit``
+
+	The environment is populated by running the script specified in the environment variable
+	``AMARANTH_ENV_ysosys_nextpnr``, if present.
+
+	Available overrides:
+		* ``nextpnr_db_dir``: overrides nextpnr-xilinx chip-db path.
+		* ``prjxray_db_dir``: overrides prjxray db path prefix.
+		* ``read_verilog_opts``: adds options for ``read_verilog`` Yosys command.
+		* ``script_after_read``: inserts commands after ``read_verilog`` in Yosys script.
+		* ``synth_opts``: adds options for ``synth_xilinx`` Yosys command.
+		* ``script_after_synth``: inserts commands after ``synth_xilinx`` in Yosys script.
+		* ``yosys_opts``: adds extra options for ``yosys``.
+		* ``nextpnr_opts``: adds extra options for ``nextpnr-xilinx``.
+		* ``fasm2frames_opts``: adds extra options for ``fasm2frames``.
+		* ``xc7frames2bit_opts``: adds extra options for ``xc7frames2bit``.
```

Comment:
Please add a new line here so the end of comment has a blank line between it and the last bits of text